lib/src/postgrest_query_builder.dart

```diff
@@ -26,12 +26,26 @@ class PostgrestQueryBuilder extends PostgrestBuilder {
   /// Performs horizontal filtering with SELECT.
   ///
   /// ```dart
-  /// postgrest.from('users').select('id, messages');
+  /// postgrest.from('users').select<PostgrestList>('id, messages');
   /// ```
-  PostgrestFilterBuilder select([
+  ///
+  /// ```dart
+  /// postgrest.from('users').select<PostgrestListResponse>('id, messages', FetchOptions(count: CountOption.exact));
+  /// ```
+  /// By setting [FetchOptions.count] to non null or [FetchOptions.forceResponse] to `true`, the return type is `PostgrestResponse<T>`.
+  ///
+  /// Allowed types for [R] are:
+  /// - [PostgrestList]
+  /// - [PostgrestMap]
+  /// - [PostgrestMap?]
+  /// - [PostgrestListResponse]
+  /// - [PostgrestMapResponse]
+  /// - [PostgrestResponse]
```

Comment:
I think it's better to avoid introducing new symbols to the users as much as possible even though it ends up being kind of long. I think it will be easier to understand. 

```suggestion
  /// Allowed types for [R] are:
  /// - [List<Map<String, dynamic>>]
  /// - [Map<String, dynamic>]
  /// - [Map<String, dynamic>?]
  /// - [PostgrestResponse<List<Map<String, dynamic>>>]
  /// - [PostgrestResponse<Map<String, dynamic>>]
  /// - [PostgrestResponse<Map<String, dynamic>?>]
```

Reply:
What was our conclusion here yesterday? I would use your suggested change, so that the actual type is not hidden.

Reply:
@Vinzent03 
Sorry, I missed this one. Let's go with the suggested change since even though it's a bit long!